src/System.Slices/System/PtrUtils.cs

```diff
@@ -43,114 +37,71 @@ sealed class PtrUtils
         /// a way that the GC will be okay with.  It yields a value of type T.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [ILSub(@"            
-            .maxstack 3
-            .locals([0] uint8 & addr)
-            ldarg.0     // load the object
-            stloc.0     // convert the object pointer to a byref
-            ldloc.0     // load the object pointer as a byref
-            ldarg.1     // load the offset
-            add         // add the offset
-            ldarg.2     // load the index
-            sizeof !!T  // load size of T
-            mul         // multiply the index and size of T
-            add         // add the result
-            ldobj !!T   // load a T value from the computed address
-            ret")]
-        public static T Get<T>(object obj, UIntPtr offset, UIntPtr index) { return default(T); }
+        public static unsafe T Get<T>(object obj, UIntPtr offset, UIntPtr index)
+        {
+            return Unsafe.Read<T>(*(byte**)Unsafe.AsPointer(ref obj) + offset.ToUInt64() + index.ToUInt32() * Unsafe.SizeOf<T>());
```

Comment:
The will do as the inliner thinks in IL. However these methods are invalid due to not being GC safe; need to use the ref return methods.